packages/build/src/lib/schema-parser/middleware/checkCache.ts

```diff
@@ -0,0 +1,82 @@
+import { ConfigurationError } from '@vulcan-sql/core';
+import { RawAPISchema, SchemaParserMiddleware } from './middleware';
+import { CACHE_METADATA_NAME } from './constants';
+import ms, { StringValue } from 'ms';
+
+export class CheckCache extends SchemaParserMiddleware {
+  public async handle(schemas: RawAPISchema, next: () => Promise<void>) {
+    // check .yaml file has cache configuration
+    const caches = schemas?.cache;
+    const metadata = schemas.metadata;
+    const isUsedCache = metadata?.[CACHE_METADATA_NAME]?.['isUsedTag'];
+    if (!isUsedCache && !caches) return next();
+
+    // throw if cache not used in .sql file
+    if (isUsedCache && !caches) {
+      throw new ConfigurationError(
+        `{% cache %} tag was used in SQL file, but the cache configurations was not found in Schema ${schemas.urlPath}.`
```

Comment:
For the `${schemas.urlPath}`, it's the same suggestion to replace `sourceName` and add a double quote on it